File: core/response.py
"""
统一响应格式工具
提供标准化的成功/错误响应包装函数。

使用约定：
  - 成功: return success(data={...})
  - 列表: return success_list(items=[...], total=100, page=1, page_size=20)
  - 错误: raise NotFound("农户不存在")

注意：现有接口保持向后兼容，新接口建议使用本工具。
"""

from fastapi.responses import JSONResponse


def success(data: object = None, message: str = "操作成功") -> dict:
    """标准成功响应"""
    return {"code": 0, "message": message, "data": data}


def success_list(
    items: list,
    total: int = None,
    page: int = None,
    page_size: int = None,
    message: str = "查询成功",
    **extra,
) -> dict:
    """分页列表响应"""
    result = {"code": 0, "message": message, "data": {"items": items}}
    if total is not None:
        result["data"]["total"] = total
    if page is not None:
        result["data"]["page"] = page
    if page_size is not None:
        result["data"]["page_size"] = page_size
    if extra:
        result["data"].update(extra)
    return result


def error_response(code: int, message: str, detail: object = None) -> JSONResponse:
    """标准错误响应（FastAPI JSONResponse，用于全局异常处理器）"""
    content = {"code": code, "message": message}
    if detail is not None:
        content["detail"] = detail
    return JSONResponse(status_code=code, content=content)


def paginated_query(
    db, query, page: int = 1, page_size: int = 20, order_by=None
) -> tuple[list, int]:
    """通用分页查询辅助"""
    total = query.count()
    if order_by is not None:
        query = query.order_by(order_by)
    items = query.offset((page - 1) * page_size).limit(page_size).all()
    return items, total

File: subsidy-electron/electron/ipc/response.ts
/**
 * IPC 响应格式 — 对应 Python 版 core/response.py
 */

export interface ApiResponse<T = unknown> {
  code: number
  data?: T
  message?: string
}

export interface PageResult<T> {
  items: T[]
  total: number
  page: number
  page_size: number
}

/**
 * 成功响应
 */
export function success<T>(data: T, message = 'ok'): ApiResponse<T> {
  return { code: 0, data, message }
}

/**
 * 列表响应
 */
export function successList<T>(items: T[], total: number, page = 1, pageSize = 20): { code: number; data: PageResult<T> } {
  return { code: 0, data: { items, total, page, page_size: pageSize } }
}

/**
 * 错误响应
 */
export function errorResponse(message: string, code = 400): ApiResponse {
  return { code, message }
}

/**
 * 从 query string 解析分页参数
 */
export function parsePagination(params: Record<string, unknown>): { page: number; pageSize: number; offset: number } {
  const page = Math.max(1, parseInt(String(params.page || '1')) || 1)
  const pageSize = Math.min(100, Math.max(1, parseInt(String(params.page_size || String(params.pageSize || '20'))) || 20))
  const offset = (page - 1) * pageSize
  return { page, pageSize, offset }
}
